axelrod/tests/unit/test_gambler.py

```diff
@@ -90,39 +94,38 @@ class TestPSOGambler(TestPlayer):
 
     def test_init(self):
         # Check for a few known keys
-        known_pairs = {('CD', 'DD', 'DD'): 0.48, ('CD', 'CC', 'DD'): 0.67}
+        known_pairs = {('CD', 'DD', 'DD'): 0.24523149, ('CD', 'CC', 'DD'): 0.}
         player = self.player()
         for k, v in known_pairs.items():
             self.assertEqual(player.lookup_table[k], v)
 
     def test_strategy(self):
         """Starts by cooperating."""
         self.first_play_test(C)
-        # Defects on the last two rounds
-        self.responses_test([C] * 197, [C] * 197, [C, D, D],
-                            tournament_length=200)
+        self.responses_test([C] * 197, [C] * 197, [C])
 
 
 # Some heads up tests for PSOGambler
-class PSOGamblervsDefector(TestHeadsUp):
+class PSOGambler2_2_2vsDefector(TestHeadsUp):
     def test_vs(self):
-        self.versus_test(axelrod.PSOGambler(), axelrod.Defector(),
+        self.versus_test(axelrod.PSOGambler2_2_2(), axelrod.Defector(),
                          [C, C, D, D], [D, D, D, D])
 
 
-class PSOGamblervsCooperator(TestHeadsUp):
+class PSOGambler2_2_2vsCooperator(TestHeadsUp):
     def test_vs(self):
-        self.versus_test(axelrod.PSOGambler(), axelrod.Cooperator(),
+        self.versus_test(axelrod.PSOGambler2_2_2(), axelrod.Cooperator(),
                          [C, C, C, C], [C, C, C, C])
 
 
-class PSOGamblervsTFT(TestHeadsUp):
+class PSOGambler2_2_2vsTFT(TestHeadsUp):
     def test_vs(self):
-        self.versus_test(axelrod.PSOGambler(), axelrod.TitForTat(),
+        self.versus_test(axelrod.PSOGambler2_2_2(), axelrod.TitForTat(),
                          [C, C, C, C], [C, C, C, C])
 
 
-class PSOGamblervsAlternator(TestHeadsUp):
+class PSOGambler2_2_2vsAlternator(TestHeadsUp):
     def test_vs(self):
-        self.versus_test(axelrod.PSOGambler(), axelrod.Alternator(),
-                         [C, C, D, D, D, D], [C, D, C, D, C, D])
+        axelrod.seed(10)
+        self.versus_test(axelrod.PSOGambler2_2_2(), axelrod.Alternator(),
+                         [C, C, C, C, C, C, C], [C, D, C, D, C, D, C])
```

Comment:
Could we also have tests for the other new gamblers please (specific plays and potentially checks for structure of underlying table):

- `Mem1`
- `1_1_1`
- `2_2_2`
- `2_2_2_noise05`